### ISSUE TITLE
System metrics support

### ISSUE THREAD

**Issue by kerthcet** [kubernetes-sigs/lws#87]
**What would you like to be added**:

We should add some metrics for tracking the health of the system, some of them like:
- Rolling update duration
- Recreate group times
- HPA related? Like scaling up/down times & duration?

Please add more if you think so.

**Why is this needed**:

Providing insight of the system

**Completion requirements**:

This enhancement requires the following artifacts:

- [ ] Design doc
- [ ] API change
- [x] Docs update

The artifacts should be linked in subsequent comments.

**Comment by liurupeng** [comment]
+1 this should be pretty useful. How about "latency for pod be scheduled and become ready"?

**Comment by liurupeng** [comment]
> Or do you mean Replica? Pod seems not that important.

yep, Replica makes sense

**Comment by googs1025** [comment]
"Hey, do we still need this feature? I'm not sure why the PR got closed back then and nothing happened afterwards. If it is still needed, I can take a shot at it." @kerthcet @liurupeng @Edwinhr716 
Or, it would be great if Edwinhr716 is willing to re-promote this PR

**Comment by Edwinhr716** [comment]
The PR got closed due to lack of clarity on what each metric was measuring. If we want to continue with the issue, it would be good to agree on what metrics we want to be added and what each metric entails

**Comment by kerthcet** [comment]
Yes, we do. I'll categorize them once I had time and submit a pr as a demo.

**Comment by Edwinhr716** [comment]
@kerthcet any update on this? I think it will be useful to have for 0.7.0. I'm happy to help in the discussion on what metrics are needed

**Comment by kerthcet** [comment]
Let's target for v0.7.0,  I'll back to this after kubecon EU.

**Comment by Edwinhr716** [comment]
Since we are including this as part of the 0.7.0, would be good to define the metrics that we want to be included. 

Based on https://github.com/kubernetes-sigs/lws/pull/97/files#r1569659033,
> I don't think we need such a metric. The time it takes for the pod to become ready is not a characteristic of the lws controller.

we don't want to do any metric that depends on how long it takes for pods to be ready. So we shouldn't have `rolling_update_duration` either. 

I think `recreate_group_times` is a metric worth having. 

Using the statefulset metrics as a pointer https://github.com/kubernetes/kube-state-metrics/blob/main/docs/metrics/workload/statefulset-metrics.md, having something like `lws_replicas_updated` might be useful? There's also `statefulset_unavailability_violation` which I'm adding to the statefulset controller, we could add an equivalent for lws as well. 

Though is it worth it to have an lws equivalent when getting the metric from the leader statefulset will yield the same result?
